File: Everly Precia Suresh/Arrays/right_rotate.cpp
#include<iostream>
using namespace std;
void right_rotate(int arr[],int n){
   int last=arr[n-1];
   for(int i=n-1;i>=0;i--){
       arr[i]=arr[i-1];
   }
   arr[0]=last;
}
int main(){
    int n;
     
    cout<<"Enter number of elements : ";
    cin>>n;
    int ar[n];
    cout<<"Enter Elements: ";

    for(int i=0;i<n;i++){ //read
        cin>>ar[i];
    }
    cout<<"Array inputted successfully"<<endl;
    int d;
    cout<<"Enter times to right rotate: ";
    cin>>d;
    int t=d;
    while(d--){
        right_rotate(ar,n);
    }
     cout<<"Array after righ rotation "<<t<<" times: ";
    for(int i=0;i<n;i++){
        cout<<ar[i]<<" ";
    }
    return 0;
}

File: Arnav Sharma/Assignment-1/right_rotate.java
package com.array;

public class right_rotate {
    public static void main(String[] args) {

        int [] arr = new int [] {1, 2, 3, 4, 5};
        int n = 3;

        System.out.println("Original array: ");
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
        for(int i = 0; i < n; i++){
            int j, last;
            last = arr[arr.length-1];

            for(j = arr.length-1; j > 0; j--){
                arr[j] = arr[j-1];
            }
            arr[0] = last;
        }

        System.out.println();

        System.out.println("Array after right rotation: ");
        for(int i = 0; i< arr.length; i++){
            System.out.print(arr[i] + " ");
        }
    }
}
